Do not ignore alternative generation strategies for identifiers

spinx opened the pull request:
Fix for https://github.com/trappar/AliceGeneratorBundle/issues/3

Commit:
Do not ignore alternative generation strategies for identifiers

trappar reviewed (changes requested):
It appears that you can access what auto generation strategy is used at `$classMetadata->generatorType` and compare that to `Doctrine\ORM\Mapping\ClassMetadataInfo::GENERATOR_TYPE_NONE`. Doing it that way seems preferable since it wouldn't involve any additional use of the annotation reader.

Commit:
generator none fix and tests

spinx commented:
I did that because there's an option to use composite keys and the generator type is only set for the entity. 

I can see now that Doctrine doesn't even support this, as it only works on MySQL MyISAM anyways, so your suggestion is on point. 

I've fixed it, made sure it works correctly with composite keys and added tests.

trappar commented:
Thanks so much for contributing! I'll get this tagged shortly 😄
